src/heap_ptmalloc.cpp

```diff
@@ -2185,3 +2199,7 @@ memcpy_field_value(struct value *val, const char *fieldname, char *buf,
 	}
 	return true;
 }
+
+class PtMallocHeap: public CoreAnalyzerHeapInterface {
```

Comment:
This line will be removed. In the beginning, I would like to use the c++ class to define the interface. Later I thought the gdb is written only in C and the class cannot be used. So I switch to using the C struct.  So the current version is using the C struct. This line is obsolete. But it turns out the gdb can use c++ code even if the file is with `.c` suffix. It is kind of strange, I haven't fully understood how it works, so I will stick to the C struct interface.
cc @yanqi27

Reply:
Probably around gdb v9, it switches to c++ compiler. It probably compiles all sources with c++ regardless to its suffix.

Reply:
If cpp could be used, I'm thinking switching to use C++ abstract class to define the interface.